app/src/main/java/at/tugraz/ist/guessingwords/data/service/WordService.kt

```diff
@@ -48,4 +48,46 @@ open class WordService(private val context : Context) {
             callback.whenReady(word)
         }
     }
+
+    /**
+     * Creates an In Memory Instance for merging Word Pools
+     * Populates it with local persistent Words
+     * @param callback
+     */
+    open fun createNewMultiplayerWordPool(callback: Callback<List<Long>>) {
+        thread {
+            val localWords = GWDatabase.getInstance(context).wordDao().getAll()
+            val dbTemp = GWDatabase.getInMemoryInstance(context)
```

Comment:
We cannot be sure if `removeMultiplayerWordPool()` was actually called before creating a New Multiplayer Word Pool.
So either call it before accessing the InMemoryInstance or clear it manually first.

In any case, the InMemoryInstance should be empty here, but this is currently not guaranteed